### PR TITLE
chore: Update dependencies and add linting

### PR DESCRIPTION
**Pull request by ffflorian**

### CODE REVIEW

File: .huskyrc.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  hooks: {
+    'pre-commit': 'lint-staged',
+  },
+};
```
Comment (lipis) [inline .huskyrc.js]: is it really necessary to have it separate file instead of including it in package.json?

Comment (lipis) [reply .huskyrc.js]: We removed prettierrc and added into package.json.. why not doing the same with this one?

Comment (ffflorian) [reply .huskyrc.js]: Probably not.

File: package.json [hunk 2]
```diff
@@ -39,20 +45,39 @@
...
     "dist": "yarn clear && yarn build",
+    "fix": "yarn fix:other && yarn fix:ts",
+    "fix:other": "yarn prettier --write",
+    "fix:ts": "yarn lint:ts --fix",
+    "lint": "yarn lint:other && yarn lint:ts",
+    "lint:other": "yarn prettier --list-different",
+    "lint:ts": "tslint --project tsconfig.json",
     "postinstall": "node-gyp rebuild",
-    "test": "exit 0"
+    "prettier": "prettier \"**/*.{json,md,css}\"",
```
Comment (lipis) [inline package.json]:
```suggestion
    "prettier": "prettier \"**/*.{json,md,css,yml}\"",
```

File: package.json [hunk 3]
```diff
@@ -39,20 +45,39 @@
     "binding",
     "c"
   ],
+  "license": "GPL-3.0",
+  "lint-staged": {
+    "**/*.ts": [
+      "tslint --project tsconfig.json --fix",
+      "git add"
+    ],
+    "*.{json,md,css}": [
```
Comment (lipis) [inline package.json]:
```suggestion
    "*.{json,md,css,yml}": [
```

**Pull request merged**